tests/components/bayesian/test_binary_sensor.py

```diff
@@ -331,6 +325,69 @@ async def test_sensor_value_template(hass: HomeAssistant) -> None:
     assert state.state == "off"
 
 
+async def test_mixed_states(hass: HomeAssistant) -> None:
+    """Test sensor on probability threshold limits."""
+    config = {
+        "binary_sensor": {
+            "name": "should_HVAC",
+            "platform": "bayesian",
+            "observations": [
+                {
+                    "platform": "template",
+                    "value_template": "{{states('sensor.guest_sensor') != 'off'}}",
+                    "prob_given_true": 0.3,
+                    "prob_given_false": 0.15,
+                },
+                {
+                    "platform": "state",
+                    "entity_id": "sensor.anyone_home",
+                    "to_state": "on",
+                    "prob_given_true": 0.6,
+                    "prob_given_false": 0.05,
+                },
+                {
+                    "platform": "numeric_state",
+                    "entity_id": "sensor.temperature",
+                    "below": 24,
+                    "above": 19,
+                    "prob_given_true": 0.1,
+                    "prob_given_false": 0.6,
+                },
+            ],
+            "prior": 0.3,
+            "probability_threshold": 0.5,
+        }
+    }
+    assert await async_setup_component(hass, "binary_sensor", config)
+    await hass.async_block_till_done()
+
+    hass.states.async_set("sensor.guest_sensor", "UNKNOWN")
+    hass.states.async_set("sensor.anyone_home", "on")
+    hass.states.async_set("sensor.temperature", 15)
+
+    await hass.async_block_till_done()
+
+    state = hass.states.get("binary_sensor.should_HVAC")
+
+    assert set(state.attributes.get("occurred_observation_entities")) == {
+        "sensor.anyone_home",
+        "sensor.temperature",
+    }
+    template_obs = {
+        "platform": "template",
+        "value_template": "{{states('sensor.guest_sensor') != 'off'}}",
+        "prob_given_true": 0.3,
+        "prob_given_false": 0.15,
+        "observed": True,
+    }
+    assert template_obs in state.attributes.get("observations")
+
+    assert abs(0.95857988 - state.attributes.get("probability")) < 0.01
+    # Calculated using bayes theorum where P(A) = 0.3, P(B|A) = 0.3 , P(B|notA) = 0.15 = 0.46153846
+    # Step 2 0.91139240
```

Comment:
lets also include the formula for step 2 and 3

Reply:
fixed in next commit